### PR TITLE
fix: Fix compiler crash when certain operators used on types #3

### PR DESCRIPTION
**Pull request by romdotdog**
This is a continuation of https://github.com/AssemblyScript/assemblyscript/pull/2293#discussion_r873211983.

- [x] I've read the contributing guidelines
- [x] I've added my name and email to the NOTICE file

### CODE REVIEW

**Review by dcodeIO — APPROVED** [review]

File: tests/compiler/unary-errors.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+const g = (Array++);
+const h = (++Array);
+// const i = (Array += 1);
+
+const j = (Array--);
+const k = (--Array);
+// const l = (Array -= 1);
+
+ERROR("EOF");
```
Comment (MaxGraey) [inline tests/compiler/unary-errors.ts:38]: I guess all this test may be simplified just to this:
```ts
/* eslint-disable no-class-assign */
/* eslint-disable no-global-assign */

class Foo {}

const a = (Foo--);
const b = (++Foo);

RROR("EOF"); 
```

Comment (romdotdog) [reply tests/compiler/unary-errors.ts:38]: I think more test cases could be useful for the scenario in which someone breaks specific internals and one of the cases fails.

Comment (MaxGraey) [reply tests/compiler/unary-errors.ts:38]: `Array` is the same class as Foo and will never have overload ops like `--` or `++`. So I don't see a reason test it. Tests with assignments it seems also covers tests without assignments so it also redundant

**Review by MaxGraey — APPROVED** [review]

**Comment by MaxGraey** [comment]
Thanks!

**Pull request merged**